ffmpeg-full broken on Darwin (libglslang not found in configure step)

**Describe the bug**
Building ffmpeg-full on Darwin fails in the configure step, with the following error message:

```
configure flags: --disable-static --prefix=/nix/store/dhrqbjnamwz1xw3p9h463qbmmc76yq40-ffmpeg-full-4.4 --target_os=darwin --arch=x86_64 --enable-gpl --enable-version3 --disable-nonfree --enable-shared --enable-pic --cc=clang --disable-small --enable-runtime-cpudetect --disable-lto --enable-gray --enable-swscale-alpha --enable-hardcoded-tables --enable-safe-bitstream-reader --enable-pthreads --disable-w32threads --disable-os2threads --enable-network --enable-pixelutils --enable-ffmpeg --enable-ffplay --enable-ffprobe --enable-avcodec --enable-avdevice --enable-avfilter --enable-avformat --enable-avresample --enable-avutil --enable-postproc --enable-swresample --enable-swscale --enable-doc --disable-htmlpages --enable-manpages --disable-podpages --disable-txtpages --enable-bzlib --enable-libcelt --enable-libdav1d --disable-libfdk-aac --disable-libflite --enable-fontconfig --enable-libfreetype --enable-frei0r --enable-libfribidi --enable-libgme --enable-gnutls --enable-libgsm --enable-ladspa --enable-libmp3lame --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --disable-libdc1394 --enable-iconv --disable-libjack --disable-libmfx --disable-libmodplug --enable-libmysofa --enable-libopus --enable-librsvg --enable-libsrt --enable-libssh --enable-libtheora --disable-libv4l2 --disable-vaapi --enable-vdpau --enable-libvorbis --enable-libvmaf --enable-libvpx --enable-libwebp --enable-xlib --enable-libxcb --enable-libxcb-shm --enable-libxcb-xfixes --enable-libxcb-shape --enable-lzma --disable-nvenc --enable-openal --enable-libopencore-amrnb --disable-opengl --enable-libopenjpeg --disable-openssl --disable-libpulse --disable-librav1e --enable-libsvtav1 --enable-librtmp --enable-sdl2 --enable-libsoxr --enable-libspeex --disable-libvidstab --enable-libvo-amrwbenc --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzmq --enable-zlib --disable-vulkan --enable-libglslang --disable-debug --enable-optimizations --disable-extra-warnings --disable-stripping
ERROR: libglslang not found

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.
builder for '/nix/store/pd9ngrphk41y9chssqsg3bxix072bjyv-ffmpeg-full-4.4.drv' failed with exit code 1
error: build of '/nix/store/24312m1qrv45r0zgmb1b6nyaj7x6gm88-bash-interactive-4.4-p23.drv', '/nix/store/pd9ngrphk41y9chssqsg3bxix072bjyv-ffmpeg-full-4.4.drv' failed
```

**To Reproduce**
Steps to reproduce the behavior:
```
nix-shell -p ffmpeg-full -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/a39809ade0121f374126b5b499173930672b3736.tar.gz
```

**Expected behavior**
ffmpeg-full should build, and nix-shell should display a shell prompt.

**Additional context**
This library was added to the ffmpeg-full build inputs in #120482.  @primeos suggests that it might actually be `libglslang` which is broken on Darwin, causing the headers not to be found by the `ffmpeg-full` build.

**Notify maintainers**

glslang: @ralith
ffmpeg-full: @codyopel 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- nixpkgs.ffmpeg-full
# a list of nixos modules affected by the problem
module:
```

The first step to fix this is to pass `--cxx=clang++` in addition to `--cc=clang` we already do. It looks like it was trying to invoke g++ for the check.

Now seeing the following linker error locally:

```
LD      libavfilter/libavfilter.7.dylib
Undefined symbols for architecture x86_64:
  "glslang::GlslangToSpv(glslang::TIntermediate const&, std::__1::vector<unsigned int, std::__1::allocator<unsigned int> >&, spv::SpvBuildLogger*, glslang::SpvOptions*)", referenced from:
      _glslang_compile in glslang.o
  "glslang::FinalizeProcess()", referenced from:
      _glslang_uninit in glslang.o
  "glslang::InitializeProcess()", referenced from:
      _glslang_init in glslang.o
  "glslang::TShader::getInfoLog()", referenced from:
      _glslang_compile in glslang.o
  "glslang::TShader::setStrings(char const* const*, int)", referenced from:
      _glslang_compile in glslang.o
  "glslang::TShader::parse(TBuiltInResource const*, int, EProfile, bool, bool, EShMessages, glslang::TShader::Includer&)", referenced from:
      _glslang_compile in glslang.o
  "glslang::TShader::TShader(EShLanguage)", referenced from:
      _glslang_compile in glslang.o
  "glslang::TProgram::getInfoLog()", referenced from:
      _glslang_compile in glslang.o
  "glslang::TProgram::link(EShMessages)", referenced from:
      _glslang_compile in glslang.o
  "glslang::TProgram::TProgram()", referenced from:
      _glslang_compile in glslang.o
  "vtable for __cxxabiv1::__class_type_info", referenced from:
      typeinfo for glslang::TShader::Includer in glslang.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for __cxxabiv1::__si_class_type_info", referenced from:
      typeinfo for glslang::TShader::ForbidIncluder in glslang.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "operator delete(void*)", referenced from:
      _glslang_compile in glslang.o
      glslang::TShader::ForbidIncluder::~ForbidIncluder() in glslang.o
  "operator new(unsigned long)", referenced from:
      _glslang_compile in glslang.o
  "___gxx_personality_v0", referenced from:
      _glslang_compile in glslang.o
      Dwarf Exception Unwind Info (__eh_frame) in glslang.o
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [ffbuild/library.mak:103: libavfilter/libavfilter.7.dylib] Error 1
```

--- a/pkgs/development/libraries/ffmpeg-full/default.nix
+++ b/pkgs/development/libraries/ffmpeg-full/default.nix
@@ -272,7 +272,6 @@ stdenv.mkDerivation rec {
     # On some ARM platforms --enable-thumb
     "--enable-shared"
     (enableFeature true "pic")
-    (if stdenv.cc.isClang then "--cc=clang" else null)
     (enableFeature smallBuild "small")
     (enableFeature runtimeCpuDetectBuild "runtime-cpudetect")
     (enableFeature enableLto "lto")
@@ -404,7 +403,7 @@ stdenv.mkDerivation rec {
     (enableFeature (zeromq4 != null) "libzmq")
     (enableFeature (zlib != null) "zlib")
     (enableFeature (isLinux && vulkan-loader != null) "vulkan")
-    (enableFeature (glslang != null) "libglslang")
+    (enableFeature (isLinux && vulkan-loader != null && glslang != null) "libglslang")
     #(enableFeature (zvbi != null && gplLicensing) "libzvbi")
     /*
      * Developer flags
@@ -416,6 +415,9 @@ stdenv.mkDerivation rec {
   ] ++ optionals (stdenv.hostPlatform != stdenv.buildPlatform) [
     "--cross-prefix=${stdenv.cc.targetPrefix}"
     "--enable-cross-compile"
+  ] ++ optionals stdenv.cc.isClang [
+    "--cc=clang"
+    "--cxx=clang++"
   ];
 
   nativeBuildInputs = [ addOpenGLRunpath perl pkg-config texinfo yasm ];
@@ -426,12 +428,12 @@ stdenv.mkDerivation rec {
     libogg libopus librsvg libssh libtheora libvdpau libvorbis libvpx libwebp libX11
     libxcb libXv libXext xz openal openjpeg libpulseaudio rav1e svt-av1 rtmpdump opencore-amr
     samba SDL2 soxr speex srt vid-stab vo-amrwbenc x264 x265 xavs xvidcore
-    zeromq4 zlib glslang
+    zeromq4 zlib
   ] ++ optionals openglExtlib [ libGL libGLU ]
     ++ optionals nonfreeLicensing [ fdk_aac openssl ]
     ++ optional ((isLinux || isFreeBSD) && libva != null) libva
     ++ optional (!isAarch64 && libvmaf != null && version3Licensing) libvmaf
-    ++ optionals isLinux [ alsaLib libraw1394 libv4l vulkan-loader ]
+    ++ optionals isLinux [ alsaLib libraw1394 libv4l vulkan-loader glslang ]
     ++ optional (isLinux && !isAarch64 && libmfx != null) libmfx
     ++ optional nvenc nv-codec-headers
     ++ optionals stdenv.isDarwin [ Cocoa CoreServices CoreAudio AVFoundation

